### PR TITLE
Refactor team stats file and broke most variables out into helper met…

### PR DESCRIPTION
**Pull request by araspaas**
Refactor team stats file and broke most variables out into helper methods. took a second off of the runtime for me

### CODE REVIEW

File: lib/team_stats.rb [hunk 1]
```diff
@@ -70,31 +75,55 @@ def find_all_games_by_given_team_id(team_id)
     @games.find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
   end
 
-  def favorite_opponent(team_id)
-    find_game_ids = find_all_games_by_given_team_id(team_id).map {|game| game.game_id}
-    find_given_ids_opponents = @game_teams.find_all {|game| find_game_ids.include?(game.game_id) && game.team_id != team_id}
-    sorted_by_team_id = find_given_ids_opponents.group_by {|game| game.team_id}
-    result = sorted_by_team_id.transform_values do |game|
+  def find_game_ids(team_id)
+    find_all_games_by_given_team_id(team_id).map {|game| game.game_id}
```
Comment (dcoleman21) [inline lib/team_stats.rb:79]: Nice job on these

**Review by robertorodriguez12 — COMMENTED** [review]
I like what you did here

File: lib/team_stats.rb [hunk 2]
```diff
@@ -70,31 +75,55 @@ def find_all_games_by_given_team_id(team_id)
...
-      count_of_wins = game.find_all do |game|
-        (game.result == "WIN")
-      end.count
-      (count_of_wins.to_f / game.count).round(2)
-    end
-    the_team = result.max_by {|_, ratio| ratio}.first
+    find_game_ids = find_game_ids(team_id)
+    find_given_ids_opponents = find_given_ids_opponents(team_id)
+    sorted_by_team_id = sorted_by_team_id(team_id)
+    results_where_team_lost = results_where_team_lost(team_id)
```
Comment (dcoleman21) [inline lib/team_stats.rb:125]: This is way smaller

**Review by robertorodriguez12 — APPROVED** [review]

**Review by dcoleman21 — COMMENTED** [review]
Nice job Austin

**Pull request merged**